### PR TITLE
feat(main): add --watch-namespace to scope the manager cache

### PR DESCRIPTION
**Pull request by kjvalencik**
## Motivation

The manager's controller-runtime cache defaults to cluster-wide watches for every secondary resource it owns (Pods, PVCs, Services, Jobs, PDBs). On large clusters, caching every pod in the cluster can OOM-kill the manager against the chart's default 128Mi limit — even when only a handful of namespaces contain etcd clusters. `docs/installation.md` previously called this out as an unexposed follow-up.

Related: #199

## Changes

- New `--watch-namespace` flag, defaulting to `$WATCH_NAMESPACE` (operator-sdk convention): a comma-separated namespace list wired into the manager's `Cache.DefaultNamespaces`. Empty/unset keeps today's watch-everything behavior.
- Chart: `manager.watchNamespaces` (list) renders the `WATCH_NAMESPACE` env var onto the manager container. Unset by default.
- Docs: installation.md RBAC section and chart-values table updated.

## RBAC

Intentionally untouched — the existing ClusterRole authorizes the scoped watches. Since every rule in the manager role covers namespaced resources, this change also makes it *possible* to run with per-namespace `Role`s instead of the ClusterRole (with the metrics proxy disabled, as its TokenReview/SubjectAccessReview permissions are cluster-scoped) — left as a future improvement.

## Testing

- Unit tests for the namespace-list parsing: empty, whitespace-only, and separator-only inputs yield the zero `cache.Options`, keeping the unset path identical to the previous behavior.
- `make test` passes; `helm template`/`helm lint` verified with and without the value set.

### CODE REVIEW

File: charts/etcd-operator/templates/deployment.yaml [hunk 1]
```diff
@@ -59,6 +59,10 @@ spec:
         # the tag is always v<spec.version>.
         - name: ETCD_IMAGE_REPOSITORY
           value: {{ .Values.etcdImage.repository | quote }}
+        {{- with .Values.manager.watchNamespaces }}
+        - name: WATCH_NAMESPACE
+          value: {{ join "," . | quote }}
+        {{- end }}
```
[the remark this answers is not in the document]
Comment (kjvalencik) [reply charts/etcd-operator/templates/deployment.yaml:65]: This is _inaccurate_ because `join` stringifies scalars, but I made the change anyway because this is cleaner.

**Review by androndo — APPROVED** [review]
**LGTM**

Reviewed at head `5b7b50c` against `main` (merge base `9eb1ec1`). A tidy, well-scoped change that does exactly what the PR body claims: adds a `--watch-namespace` flag (defaulting to `$WATCH_NAMESPACE`, operator-sdk convention) wired into `Cache.DefaultNamespaces`, plus a chart value `manager.watchNamespaces` and a docs update.

What I verified:

- **Correctness of cache scoping.** The one real risk with `DefaultNamespaces` is a cached read landing in a namespace outside the watch set. I traced every cached `Get`/`List` in `controllers/` — all are scoped to the CR's own namespace (`cluster.Namespace`, `member.Namespace`, `snapshot.Namespace`), and the associated Pods/PVCs/Services/Secrets all live in that same namespace. No cross-namespace cached reads, so scoping to the watched namespaces is complete: nothing the operator needs falls outside the cache. Cluster-scoped/discovery paths (`detectCertManager` via discovery client, leader election via direct client) don't go through the manager cache, so they're unaffected. controller-runtime is v0.21.0, where `DefaultNamespaces` restricts only namespaced resources — consistent with the above.

- **No regression on the default path.** Empty/unset input returns `cache.Options{}`, byte-identical to the previous behavior (no `Cache` field was set before, i.e. the zero value). Confirmed by test and by `helm template` with no value producing zero `WATCH_NAMESPACE` env vars.

- **Helm template edge cases.** `--set 'manager.watchNamespaces={a,b}'` → `value: "a,b"`; string value → verbatim; unset (`[]`, falsy under `{{- with }}`) → env var omitted. The `kindIs "slice"` branch handles both list and string inputs. All three verified via `helm template`.

- **Tests.** `main_test.go` covers the new `watchNamespaceCacheOptions` thoroughly — valid inputs (single, multiple) AND the invalid/degenerate paths (empty, whitespace-only, separator-only, mixed empty entries), asserting the unset path stays identical to the old config. `go test .` passes, `go vet` clean.

- **Documentation.** The change correctly replaces the now-stale `docs/installation.md` paragraph that said single-namespace scoping "is not currently exposed" — the exact follow-up this PR closes. New RBAC text and the values-table row match the implementation, and the `[RBAC](#rbac)` link resolves to the `## RBAC` header.

Non-blocking cosmetic note (recommended, not required):
- Flag is singular `--watch-namespace` while the chart value is plural `manager.watchNamespaces`. This follows the operator-sdk `WATCH_NAMESPACE`-holds-a-list convention, so it's defensible, but the singular/plural split is a minor readability wrinkle.
